### PR TITLE
Discovery service authentication

### PR DESCRIPTION
**Pull request by taban03**
Documentation to describe authentication in Discovery Service for HTTP and HTTPS

### CODE REVIEW

File: docs/extend/extend-apiml/api-mediation-security.md [hunk 1]
```diff
@@ -260,12 +261,26 @@ The Discovery Service has the following types of users that require authenticati
 
 - **Administrators and developers who need to log in to the homepage of the Discovery Service**
   
-    The access is protected by a certificate.
+    - HTTP mode: the access is protected by basic Eureka authentication.
+    
+    - HTTPS mode: the access is protected by mainframe credentials (basic or token) or x509 certificate.
```
Comment (jandadav) [inline docs/extend/extend-apiml/api-mediation-security.md]:
```suggestion
    - HTTPS mode: the access is protected by mainframe credentials (basic or token) or client certificate.
```

File: docs/extend/extend-apiml/api-mediation-security.md [hunk 2]
```diff
@@ -260,12 +261,26 @@ The Discovery Service has the following types of users that require authenticati
...
+    The `application/**` endpoint is protected by mainframe credentials (basic or token) and the `discovery/**` endpoint is protected by mainframe credentials (basic or token) or x509 certificate.
```
Comment (jandadav) [inline docs/extend/extend-apiml/api-mediation-security.md]: it's endpoints, not one endpoint. x509 maybe better said as client certificate

File: docs/extend/extend-apiml/api-mediation-security.md [hunk 3]
```diff
@@ -337,18 +352,7 @@ https://github.com/zowe/api-layer/blob/master/keystore/README.md#trust-certifica
...
-
-Some utilities including HTTPie require the certificate to be in PEM format. You can find it in `keystore/localhost/localhost.pem`.
-
-Since the Discovery Service is using HTTPS, your client also requires verification of the validity of its certificate. Verification is performed by trusting the local CA certificate which is store at `keystore/local_ca/localca.cer`.
-
-The following is an example of how to access Discovery Service from CLI with full certificate validation:
-
-`http --cert=keystore/localhost/localhost.pem --verify=keystore/local_ca/localca.cer -j GET https://localhost:10011/eureka/apps/`
-
+To access Discovery Service on localhost provide valid mainframe credentials or token.
```
Comment (jandadav) [inline docs/extend/extend-apiml/api-mediation-security.md]: this is ambiguous, maybe it would be more effective to plug in the security matrix table?

Comment (taban03) [reply docs/extend/extend-apiml/api-mediation-security.md]: Good idea, the table could be more clear

**Pull request merged**